### PR TITLE
feat: adjust job priority and weight through job queue

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

add support for adjusting job priority and weight through WebUI and CLI.

co-author @eecsliu 

## Commentary (optional)

For Kubernetes, use this cluster: http://34.82.124.158:8080/

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/experiment.go [hunk 1]
```diff
@@ -252,20 +253,16 @@ func (e *experiment) Receive(ctx *actor.Context) error {
...
-	case sproto.SetGroupPriority:
-		resources := e.Config.Resources()
-		resources.SetPriority(msg.Priority)
-		e.Config.SetResources(resources)
 		msg.Handler = ctx.Self()
 		ctx.Tell(e.rm, msg)
-
+	case job.SetGroupWeight:
+		if err := e.setWeight(ctx, msg.Weight); err != nil {
+			ctx.Log().WithError(err)
```
Comment (hamidzr) [inline master/internal/experiment.go:260]: we probably should change this to arrive here with an Ask ratherthan Tell so we can respond and return the error to the user.

File: master/internal/resourcemanagers/agent_resource_manager.go [hunk 2]
```diff
@@ -64,7 +64,9 @@ func (a *agentResourceManager) Receive(ctx *actor.Context) error {
 	case sproto.ResourcesReleased:
 		a.forwardToAllPools(ctx, msg)
 
-	case sproto.SetGroupMaxSlots, sproto.SetGroupWeight, sproto.SetGroupPriority:
+	case sproto.SetGroupMaxSlots, job.SetGroupWeight, job.SetGroupPriority:
+		// QUESTION do we wanna be forwarding these to all pools? they don't seem to
```
Comment (eecsliu) [inline master/internal/resourcemanagers/agent_resource_manager.go:68]: Good question - is there confirmation this is not a problem?

**Review by eecsliu — APPROVED** [review]
LGTM! Just a couple of minor questions.

File: harness/determined/common/api/bindings.py [hunk 3]
```diff
@@ -3455,38 +3455,42 @@ class v1QueueControl:
     def __init__(
         self,
         jobId: str,
+        aheadOf: "typing.Optional[str]" = None,
+        behindOf: "typing.Optional[str]" = None,
         priority: "typing.Optional[int]" = None,
         queuePosition: "typing.Optional[float]" = None,
```
[the remark this answers is not in the document]
Comment (hamidzr) [reply harness/determined/common/api/bindings.py:3461]: I left that to be removed later so until the before/after implementation is complete we have a way of adjustin the q pos

File: master/internal/resourcemanagers/agent_resource_manager.go (hunk 2 above)
Comment (hamidzr) [reply master/internal/resourcemanagers/agent_resource_manager.go:68]: I think it ends up creating groups for those jobs on all resource pools which end up doing nothing. this is existing behavior but we might need to address it when we start switching jobs between resource pools

**Comment by hamidzr** [comment]
@hkang1 could you review the WebUI portion, please?

File: webui/react/src/pages/JobQueue/JobQueue.tsx [hunk 4]
```diff
@@ -120,9 +120,9 @@ const JobQueue: React.FC = () => {
       },
       [JobAction.ManageJob]: () => setManagingJob(job),
     };
-    if (isJobOrderAvailable && job.summary.jobsAhead > 0) {
-      triggers[JobAction.MoveToTop] = () => moveJobToPosition(job.jobId, 0);
-    }
+    // if (isJobOrderAvailable && job.summary.jobsAhead > 0) {
+    //   triggers[JobAction.MoveToTop] = () => moveJobToPosition(job.jobId, 0);
+    // }
```
[the remark this answers is not in the document]
Comment (hkang1) [reply webui/react/src/pages/JobQueue/JobQueue.tsx:125]: **question:** are there any more references to `moveJobToPosition` besides this if this is being removed?

**Review by hkang1 — APPROVED** [review]
The WebUI portion LGTM! Added some questions and comments but I don't think anything is blocking.

File: webui/react/src/pages/JobQueue/ManageJob.tsx [hunk 5]
```diff
@@ -158,46 +140,29 @@ const ManageJob: React.FC<Props> = ({ onFinish, selectedRPStats, job, schedulerT
...
+          <Form.Item
+            extra="Jobs are scheduled based on priority of 1-99 with 1 being the highest."
+            label="Priority"
+            name="priority">
+            <Input addonAfter="out of 99" max={99} min={1} type="number" />
+          </Form.Item>
+        )}
+        {schedulerType === api.V1SchedulerType.KUBERNETES && (
+          <Form.Item
+            extra="Priority is an integer with 1 being the lowest priority."
```
[the remark this answers is not in the document]
Comment (hamidzr) [reply webui/react/src/pages/JobQueue/ManageJob.tsx:153]: yeah that's better

**Comment by hamidzr** [comment]
I'm trying to deploy an aws cluster off of this but run into a master health issue 🤔 
`det deploy aws up --cluster-id jq-write --max-dynamic-agents 2 --keypair XXXX --master-instance-type t2.medium --compute-agent-instance-type=t2.medium --det-version 7a4ecea98bcd694587f5dcf1976b73b0df21f61c` @eecsliu  could you try this in your account?

**Comment by hamidzr** [comment]
http://ec2-35-83-221-71.us-west-2.compute.amazonaws.com:8080

**Comment by hamidzr** [comment]
[clabot:check]

**Comment by hamidzr** [comment]
@eecsliu could you land this once the tests pass using admin privileges? I can't land it because of the clabot. (user the pr description as the commit body)

**Comment by hamidzr** [comment]
are we still using [clabot](https://clabot.github.io/#try-it-out)?

**Pull request merged**